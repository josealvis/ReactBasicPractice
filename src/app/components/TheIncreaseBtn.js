import React  from 'react';

export class TheIncreaseBtn extends React.Component{
    
    constructor(){
        super();
        this.state = {theIncreaseNum: 1};
    }

    increase(){
    this.setState({theIncreaseNum: this.state.theIncreaseNum +1});
    }

    render(){
        return(<button type="button" onClick={() => this.increase()} className="btn btn-dark">{this.props.label +' '+ this.state.theIncreaseNum}</button>);
    }
} 

