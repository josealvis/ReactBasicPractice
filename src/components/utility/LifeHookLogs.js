import React from 'react';

export class LifeHooklogs extends React.Component{
    

    render(){
        return(<div></div>);
    }
}