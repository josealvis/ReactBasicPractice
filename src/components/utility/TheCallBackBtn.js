import React from 'react';

export class TheCallBackBtn extends React.Component{
  
    constructor(props){
        super();
        this.state = { arg1 : 'ok'}
    }
    onClickBtn(){
        this.props.callback(this.state.arg1);
    }
    render(){
        return (<button type="button" className="btn btn-warning" onClick={()=> this.onClickBtn()}>The CallBack Button </button>);
    }
}
//