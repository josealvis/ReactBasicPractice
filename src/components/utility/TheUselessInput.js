import React from 'react';

export class TheUselessInput extends React.Component{
    constructor(props){
        super();
        this.state = {text:''};
    }

    onChageText(ev){
        this.setState({text: ev.target.value })
        this.props.callback(ev.target.value);
    }

    render(){
        return(<div>
            <input className="form-control" 
                   type="text" 
                   placeholder="The Useless input"
                   value={this.state.text}
                   onChange={(ev)=> this.onChageText(ev)}
                   />
            <p>{this.state.text}</p>
        </div>);
    }
} 