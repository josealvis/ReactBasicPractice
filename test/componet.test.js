import  {TheIncreaseBtn} from '../src/components/utility/TheIncreaseBtn';
import React from 'react';

import renderer from 'react-test-renderer';
// setup file
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
// test file
import { shallow, mount, render } from 'enzyme';

test('Simple enzyme test example', () => {

  const wrapper = mount(
   <h1>The Increase Button 1</h1>
  );

  const text =wrapper.text();
  expect(text).toBe('The Increase Button 1');
});

test('testing component', () => {

  const wrapper = mount(
    <TheIncreaseBtn label={'The Increase Button'}/>
  );

  const text =wrapper.text();
  expect(text).toBe('The Increase Button 1');
});


test('Click button shuld increse it ', () => {

  const wrapper = mount(
    <TheIncreaseBtn label={'The Increase Button'}/>
  );
  wrapper.simulate('click');
  const text =wrapper.text();
  expect(text).toBe('The Increase Button 2');
});


//If we think this change is correct, we can run jest with the -u flag,
// it check if the component mach with previus snapshot
describe('Todo component renders the todo correctly', () => {
  it('renders correctly', () => {

    const rendered = renderer.create(
      <TheIncreaseBtn label={'The Increase Button'}/>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});