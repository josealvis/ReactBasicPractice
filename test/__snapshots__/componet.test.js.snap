// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders the todo correctly renders correctly 1`] = `
<button
  className="btn btn-dark"
  onClick={[Function]}
  type="button"
>
  The Increase Button 1
</button>
`;
